### DISCUSSION TITLE
We fixed the heading indentation in right sidebar.

### DISCUSSION THREAD

**Discussion by mathewmitchell** [notaapp/talk#372] [Random]
Thanks for the massive improvements (at least to my eyes) with headings and indentation. Looks/works much better now in 0.40 relative to earlier versions. 

It would be nice if at some point we could designate how many heading levels are shown. I typically only need to see as far down as H3 but don't need any H4 in the sidebar. Multimarkdown Composer added this feature a few years ago and it's really nice: a custom designed within-page navigation with pre-designated levels of detail. But, realistically, the changes in 0.4 make the viewing and using of the sidebar so much better than before: thanks, thanks, thanks!

**Comment by astoilkov** [comment]
🌈 Yeey 🌈

Always awesome to see a happy user! Thanks.

I like the suggestion. If we do it, we will need to figure out a way to do it without adding a preference (as we are aggressive at providing as few options in preferences as possible).

**Comment by Arbiter313** [comment]
I agree, the heading indentation in the outline view is much clearer and am looking forward to future improvements. Maybe a way to accommodate the above request would simply be by adding the ability to fold the headings in the outline view, as is already possible in the editor.

**Comment by petrbouchal** [comment]
Thanks for moving ahead on this! Personally I find the rounded x square icon for H1 and H2 quite difficult to interpret - I would find it more visually intuitive if H2 were also indented. A setting to limit the number of levels showing would be great,

**Reply by mathewmitchell** [reply]
I understand what you are saying, but there seems to be a need to compromise somewhere. Part of what makes the indentation work much better now is that it's bigger (i.e. indented more obviously). That makes it very easy for me to follow. Seeing no indentation between H1 and H2 is much less problematic for me as long as the length of the indentation between H2 and H3 or H3 and H4 remains the as is currently. If indentation was added between H1 and H2 something would have to give (it seems to me). If the developers chose to make the right sidebar wider that could solve the problem ... and that would be fine for me.

**Reply by Arbiter313** [reply]
I think there is probably still a way to have our cake and eat it too! I agree having the rounded and square icons to represent H1 and H2 is not ideal as it adds another layer of visual friction and unconscious interpretation. But also see the issue of the length of indentation.

I think the key would be using a visual element to keep the hierarchical relationships between heading levels extremely clear. One way to do this would be by adding vertical lines between the different heading levels, as is currently seen in the editor with indented lists (making them foldable would also be great!). In this case the length of indentation would not matter as much because the vertical lines mean the relationships are impossible to mistake. This has the added advantage of unifying the design language of the right sidebar outline view with the design language of the editor.

**Reply by DutchPete** [reply]
@qazzym 
> One way to do this would be by adding vertical lines between the different heading levels

That way H2 would have to be indented anyway, which is great because despite what @mathewmitchell says, not having H2 indented does visual friction, as you aptly describe it.

Also, in similar apps that support an outline, H2 is always indented.

**Comment by Auratelience** [comment]
One other thing is when a workspace is set to use H1 as the file title, could the H1 be hidden from the right sidebar, as it becomes a bit redundant.
Thanks
